fearless/Common/DataProvider/SingleValueProviderFactory.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import RobinHood
+import IrohaCrypto
+import FearlessUtils
+
+typealias DecodedAccountInfo = ChainStorageDecodedItem<DyAccountInfo>
+
+protocol SingleValueProviderFactoryProtocol {
+    func getPriceProvider(for assetId: WalletAssetId) -> SingleValueProvider<PriceData>
+    func getAccountProvider(for address: String, runtimeServie: RuntimeCodingServiceProtocol) throws
+    -> DataProvider<DecodedAccountInfo>
+}
+
+final class SingleValueProviderFactory {
+    static let shared = SingleValueProviderFactory(facade: SubstrateDataStorageFacade.shared,
+                                                   operationManager: OperationManagerFacade.sharedManager,
+                                                   logger: Logger.shared)
+
+    private var providers: [String: WeakWrapper] = [:]
+
+    let facade: StorageFacadeProtocol
+    let operationManager: OperationManagerProtocol
+    let logger: LoggerProtocol
+
+    init(facade: StorageFacadeProtocol, operationManager: OperationManagerProtocol, logger: LoggerProtocol) {
+        self.facade = facade
+        self.operationManager = operationManager
+        self.logger = logger
+    }
+
+    private func priceIdentifier(for assetId: WalletAssetId) -> String {
+        assetId.rawValue + "PriceId"
+    }
```

Comment:
Why can't you just use assetId as identifier?

Reply:
We can but it might be other data per asset id in future

Reply:
All right